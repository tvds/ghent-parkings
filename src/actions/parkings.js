// SET_PARKINGS
export const setParkings = (parkings) => ({
  type:'SET_PARKINGS',
  parkings
});

export const startSetParkings = () => {
  return (dispatch) => {  
    return fetch("https://datatank.stad.gent/4/mobiliteit/bezettingparkingsrealtime.json")
    .then(results =>  results.json())
    .then(data => {
      dispatch(setParkings(data))
    }).catch(error => console.log(error));
  };
}

