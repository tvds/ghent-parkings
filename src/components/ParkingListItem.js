import React from 'react';
import { Link } from 'react-router-dom';

export class ParkingListItem extends React.Component {
  render() {
    return (
      <div className={this.props.active ? 'parked' : ''} >
        <Link className="list-item" to={`/details/${this.props.id}`}>
          <div>
            <h3 className="list-item__title">{this.props.description}</h3>
            {this.props.address} {this.props.active ? this.props.parkingStatus.availableCapacity + 1 : this.props.parkingStatus.availableCapacity}/{this.props.parkingStatus.totalCapacity}  Open:{this.props.parkingStatus.open + ""}
          </div>
        </Link>
        <button onClick={this.props.onPark}>Park</button>
      </div>
    );
  };

}

export default ParkingListItem;