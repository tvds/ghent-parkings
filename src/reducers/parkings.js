// Parkings Reducer

const parkingsReducerDefaultState = [];

export default (state = parkingsReducerDefaultState, action) => {
    switch(action.type) {
        case 'SET_PARKINGS':
            return action.parkings;
        default:
            return state;
    }
};