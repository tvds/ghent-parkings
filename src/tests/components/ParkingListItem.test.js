import React from 'react';
import { shallow } from 'enzyme';
import parkings from '../fixtures/parkings';
import ParkingListItem from '../../components/ParkingListItem';

test('should render ParkingListItem correctly', () => {
  const wrapper = shallow(<ParkingListItem {...parkings[0]} />);
  expect(wrapper).toMatchSnapshot();
});
