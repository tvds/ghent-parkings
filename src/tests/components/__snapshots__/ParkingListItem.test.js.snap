// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ParkingListItem correctly 1`] = `
<div
  className=""
>
  <div
    className="list-item"
  >
    <Link
      className="list-item__link"
      replace={false}
      to="/details/18408"
    >
      <div>
        <h3
          className="list-item__title"
        >
          NaN
          . 
          Sint-Michiels
        </h3>
        <div>
          Sint-Michielsplein 8
          9000 Gent
        </div>
        <div>
          Availability: 
          93
           / 
          450
           
        </div>
      </div>
    </Link>
    <button
      className="button"
    >
      Park
    </button>
  </div>
</div>
`;
